### PR TITLE
Kafka Connect: Add config to prefix the control consumer group

### PR DESCRIPTION
**Pull request by hugofriant**
I would like to control the name of the control consumer group to have more fine grained accesses when deploying the Iceberg Sink connector on my infrastructure. 

This PR aims to replace the DEFAULT_CONTROL_GROUP_PREFIX by a configurable variable

### CODE REVIEW

File: docs/docs/kafka-connect.md [hunk 1]
```diff
@@ -77,6 +77,7 @@ for exactly-once semantics. This requires Kafka 2.5 or later.
 | iceberg.table.\<table name\>.partition-by  | Comma-separated list of partition fields to use when creating the table                                          |
 | iceberg.table.\<table name\>.route-regex   | The regex used to match a record's `routeField` to a table                                                       |
 | iceberg.control.topic                      | Name of the control topic, default is `control-iceberg`                                                          |
+| iceberg.control.group-id-prefix            | Prefix for the control consuler group, default is `cg-control`                                                   |
```
Comment (jbonofre) [inline docs/docs/kafka-connect.md:80]: There's a type here: I guess you mean `consumer` (not `consuler` :)).

I would rather name this property `iceberg.control.consumer-group-id-prefix` to keep the kafka semantic.

File: kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/IcebergSinkConfig.java [hunk 2]
```diff
@@ -80,6 +80,7 @@ public class IcebergSinkConfig extends AbstractConfig {
   private static final String TABLES_SCHEMA_CASE_INSENSITIVE_PROP =
       "iceberg.tables.schema-case-insensitive";
   private static final String CONTROL_TOPIC_PROP = "iceberg.control.topic";
+  private static final String CONTROL_GROUP_PREFIX_PROP = "iceberg.control.group-id-prefix";
```
Comment (jbonofre) [inline kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/IcebergSinkConfig.java:83]: As suggested before, I would use `consumer-group-id-prefix`.

**Review by jbonofre — CHANGES_REQUESTED** [review]

**Review by jbonofre — APPROVED** [review]

**Comment by jbonofre** [comment]
@bryanck @rdblue I reviewed this PR and worked with @hugofriant on nit changes. I think this PR is ready to be merged 😄 Can you please take a look as well ? Thanks !

**Comment by jbonofre** [comment]
@RussellSpitzer I reviewed this PR and worked with @hugofriant on nit changes. I think this PR is ready to be merged 😄 Can you also please take a look (and merge) ? Thanks !

File: kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/IcebergSinkConfig.java [hunk 3]
```diff
@@ -80,6 +80,7 @@ public class IcebergSinkConfig extends AbstractConfig {
   private static final String TABLES_SCHEMA_CASE_INSENSITIVE_PROP =
       "iceberg.tables.schema-case-insensitive";
   private static final String CONTROL_TOPIC_PROP = "iceberg.control.topic";
+  private static final String CONTROL_GROUP_ID_PREFIX_PROP = "iceberg.control.consumer-group-id-prefix";
```
[the remark this answers is not in the document]
Comment (jbonofre) [reply kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/IcebergSinkConfig.java:83]: It was the first proposal, but I proposed to keep the consumer-group semantic. OK for just group.

File: kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/IcebergSinkConfig.java [hunk 4]
```diff
@@ -359,6 +366,9 @@ public String controlTopic() {
     return getString(CONTROL_TOPIC_PROP);
   }
 
+  public String prefixControlConsumerGroup() {
```
[the remark this answers is not in the document]
Comment (jbonofre) [reply kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/IcebergSinkConfig.java:369]: Again, it was the first proposal, I asked for consumer to clear with the kafka semantic.

**Review by jbonofre — APPROVED** [review]
Thanks @hugofriant !

It looks good to me.

@bryanck @RussellSpitzer ok for you guys ?

**Comment by hugofriant** [comment]
I had to fix the format @bryanck. Could you relaunch the workflows ? Thx

**Comment by hugofriant** [comment]
Hello @bryanck, can you merge the PR ? Thank you !

**Pull request merged**